src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/DotNetCoreProjectCompatibilityDetector.cs

```diff
@@ -254,28 +286,150 @@ public int OnAfterBackgroundSolutionLoadComplete()
             {
+                if (version < compatData.UnsupportedVersion)
+                {
+                    return CompatibilityLevel.Recommended;
+                }
+
                 return CompatibilityLevel.NotSupported;
             }
 
-            return CompatibilityLevel.Supported;
+            // No restrictions
+            return CompatibilityLevel.Recommended;
+        }
+
+        /// <summary>
+        /// Pings the server to download version compatibility information and stores this in a cached file in the users app data. If the cached file is
+        /// less than 24 hours old, it uses that data. Otherwise it downloads from the server. If the download fails it will use the previously cached
+        ///  file, or if that file doesn't not exist, it uses the data baked into this class
+        /// </summary>
+        private async Task<VersionCompatibilityData> GetVersionCmpatibilityDataAsync()
+        {
+            try
+            {
+                // Do we meed to update our cached data?
+                if (_curVersionCompatibilityData == null || _timeCurVersionDataLastUpdatedUtc.AddHours(CacheFileValidHours).IsLaterThan(DateTime.UtcNow))
+                {
+                    string downLoadedVersionData = null;
+
+                    // First try the cache
+                    Dictionary<Version, VersionCompatibilityData> versionCompatData = GetCompabilityDataFromCacheFile(checkTimeStamp: true);
+
+                    if (versionCompatData == null)
+                    {
+                        // Try downloading it 
+                        try
+                        {
+                            var httpClient = new DefaultHttpClient();
+                            downLoadedVersionData = await httpClient.GetStringAsync(new Uri(VersionCompatibilityFwlink)).ConfigureAwait(false);
+                            versionCompatData = VersionCompatibilityData.DeserializeVersionData(downLoadedVersionData);
+                        }
+                        catch
+                        {
+                            // If we have a cached file use it
+                            versionCompatData = GetCompabilityDataFromCacheFile(checkTimeStamp: false);
+                        }
+                    }
+
+                    if (versionCompatData != null)
+                    {
+                        // Cache the data locally
+                        if (downLoadedVersionData != null)
+                        {
+                            FileSystem.Value.WriteAllText(VersionDataCacheFile, downLoadedVersionData);
+                        }
+
+                        // First try to match exatly on our VS version and if that fails, match on just major, minor
+                        if (versionCompatData.TryGetValue(OurVSVersion, out VersionCompatibilityData compatData) || versionCompatData.TryGetValue(new Version(OurVSVersion.Major, OurVSVersion.Minor), out compatData))
```

Comment:
nit: wrap second call to newline.

Reply:
Also, what if `OurVSVersion` is null here?

Reply:
Given the whole thing is wrapped in a try catch, it will fallback to what we shipped.